ls1intum/Artemis
# [Refactor] Decouple Orion

### Checklist

### Motivation and Context
Currently Orion specific buttons and logic is spread all over Artemis. Orion should be decoupled and have its own components and services.

Also happens to fix https://github.com/ls1intum/Orion/issues/54. It was caused by a duplicated \<jhi-alert\> that got removed by the duplication removal.

### Description
This PR introduces a new extensionPoint directive to allow Orion to reuse Artemis' templates and then decouples various components. In the future the Orion components also need to be moved to their own modules and get their own routing, this will be done at a later date; this PR only aims to divide the components.

There are no functional changes, so the tests only need to ensure there is no regression. Additionally the changes have been almost exclusively visual, so the focus should be ensuring every button is still show correctly and connected to the right function.

### Steps for Testing
The following pages should be visited with an Orion client and with a normal client and then be tested for a functioning GUI.
1. The exercise details page, especially the student actions (start/resume/clone exercise)
2. The instructor course exercise overview, especially the exercise counter at the top and the programming exercise table (Orion does not show the templates or build plans and overrides the action buttons in the last row)
3. [Orion only] the Orion instructor code editor (that opens when clicking the open in Orion in the overview in point 2)
4. The assessment exercise dashboard, especially for programming exercises.

For further reference take a look at the components in app/orion, all of those got changed in some way and should be tested

**Note for code reviews**: Most code just got moved, therefore the diff looks a bit ugly. If some method shows as removed it most likely will appear somewhere else, if some method just appears, it will probably be removed somewhere else, only exception being some added tests and some removed duplication from the student-exercise-actions.

### Test Coverage
Everything in /orion: 100%

### Screenshots
#### Exercise details
Without Orion:
![grafik](https://user-images.githubusercontent.com/61357727/125708864-0a7c2b32-326c-408e-92cc-6dcceba3f8a7.png)
With Orion:
![grafik](https://user-images.githubusercontent.com/61357727/125708893-cc2d6c5f-dea8-45f8-9634-f4e54ee95382.png)

#### Exercise overview
Without Orion
![grafik](https://user-images.githubusercontent.com/61357727/125708981-7c609087-2a9b-42d5-b86b-6268574752b4.png)
With Orion:
![grafik](https://user-images.githubusercontent.com/61357727/125708962-339c62f9-ed1c-4b95-9656-46d835429298.png)

#### Code Editor
Orion only
![grafik](https://user-images.githubusercontent.com/61357727/125709112-9cc96145-ed2d-4e80-b01a-08c1ce75859e.png)

#### Assessment exercise dashboard
Without Orion
![grafik](https://user-images.githubusercontent.com/61357727/125709210-fbb31201-055f-4e44-b7a8-6816f67b2e67.png)
With Orion
![grafik](https://user-images.githubusercontent.com/61357727/125709244-75700bca-1309-4924-a7d4-36b1e5327513.png)
